src/ray/raylet/node_manager.cc

```diff
@@ -326,6 +326,30 @@ void NodeManager::Heartbeat() {
     heartbeat_data->add_resource_load_capacity(resource_pair.second);
   }
 
+  std::unordered_set<ObjectID> active_object_ids = worker_pool_.GetActiveObjectIDs();
+  int64_t max_size = RayConfig::instance().raylet_active_object_ids_size();
+  if (active_object_ids.size() <= max_size) {
+    for (const auto &object_id : active_object_ids) {
+      if (heartbeat_data->active_object_id_size() == max_size) {
```

Comment:
I don't think this `if` is necessary since we're already in the `if` branch for less than max size.